src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/ErrorNotificationExceptionHandlerTest.java

```diff
@@ -1,60 +1,83 @@
 package uk.gov.hmcts.reform.bulkscanprocessor.exceptionhandlers;
 
-import com.microsoft.azure.servicebus.IMessage;
-import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.IQueueClient;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.client.HttpStatusCodeException;
 import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationException;
-import uk.gov.hmcts.reform.bulkscanprocessor.model.ErrorNotificationMessageWrapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus.MessageAutoCompletor;
 
 import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.verifyNoMoreInteractions;
 
+@RunWith(MockitoJUnitRunner.class)
 public class ErrorNotificationExceptionHandlerTest {
 
-    private static final IMessage MESSAGE = new Message("content");
+    @Mock
+    private IQueueClient client;
 
-    private static final ErrorNotificationExceptionHandler HANDLER = new ErrorNotificationExceptionHandler();
+    private ErrorNotificationExceptionHandler handler;
+
+    private static final UUID LOCK_TOKEN = UUID.randomUUID();
+
+    private static final CompletableFuture<Void> COMPLETED_FUTURE = CompletableFuture.completedFuture(null);
+
+    @Before
+    public void setUp() {
+        handler = new ErrorNotificationExceptionHandler(new MessageAutoCompletor(client));
+    }
 
     @Test
     public void should_mark_for_acknowledgement_when_no_exception_sent_to_handle() {
-        ErrorNotificationMessageWrapper wrapper = HANDLER.handle(MESSAGE, null);
+        given(client.completeAsync(any(UUID.class))).willReturn(COMPLETED_FUTURE);
 
-        assertThat(wrapper.isAssignedForDeadLettering()).isFalse();
-        assertThat(wrapper.isCompletedAcknowledgement()).isTrue();
+        CompletableFuture<Void> handled = handler.handle(LOCK_TOKEN, null);
+
+        assertThat(handled.join()).isNull();
```

Comment:
In the acknowledgement test and dead letter test it would be worth checking if we called complete/deadLetter. We probably won't be able to test it at a higher level.

Reply:
we define in the given/state. otherwise tests will fail. verifying interaction would be putting more butter on top

Reply:
If we commented out the call to `MessageAutoCompletor.complete` in the app code, would this test fail?

Reply:
I won't fail

P.S. There's `MessageAutoCompletorTest` which does what you are after

Reply:
Yes, those are proper tests. Here I was thinking about verifying if we're calling those two AutoCompletor's methods (which are tested) where we're expected to.